pkg/exporter/direct_flp.go

```diff
@@ -0,0 +1,44 @@
+package exporter
+
+import (
+	"fmt"
+
+	flpconfig "github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/ingest"
+	"github.com/netobserv/netobserv-ebpf-agent/pkg/flow"
+	"gopkg.in/yaml.v2"
+)
+
+// DirectFLP flow exporter
+type DirectFLP struct {
+	*ingest.InProcess
+}
+
+func StartDirectFLP(jsonConfig string) (*DirectFLP, error) {
+	var cfg flpconfig.ConfigFileStruct
+	// Note that, despite jsonConfig being json, we use yaml unmarshaler because the json one
+	// is screwed up for HTTPClientConfig in github.com/prometheus/common/config (used for Loki)
+	// This is ok as YAML is a superset of JSON.
+	// E.g. try unmarshaling `{"clientConfig":{"authorization":{}}}` as a api.WriteLoki
+	// See also https://github.com/prometheus/prometheus/issues/11816
+	if err := yaml.Unmarshal([]byte(jsonConfig), &cfg); err != nil {
+		return nil, fmt.Errorf("failed to read config: %w", err)
+	}
+
+	flp, err := pipeline.StartFLPInProcess(&cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize pipeline %w", err)
+	}
+
+	return &DirectFLP{InProcess: flp}, nil
+}
+
+// ExportFlows accepts slices of *flow.Record by its input channel, converts them
+// to *pbflow.Records instances, and submits them to the collector.
+func (d *DirectFLP) ExportFlows(input <-chan []*flow.Record) {
+	for inputRecords := range input {
+		pbRecords := flowsToPBNoChunk(inputRecords)
+		d.Write(pbRecords)
```

Comment:
This step can be removed if we let the new FLP pipe process ebpf record directly not protobuf records

Reply:
done via [bcccbb7](https://github.com/netobserv/netobserv-ebpf-agent/pull/223/commits/bcccbb7dfca1b8ec9f1e8940911dce6c59376660)